ocs-api/import-owner-private-keys2.sh

```diff
@@ -0,0 +1,300 @@
+	exit
+  fi
+fi
+
+
+wait
+
+
+#============================FUNCTIONS=================================
+
+chk() {
+    local exitCode=$1
+    local task=$2
+    local dontExit=$3   # set to 'continue' to not exit for this error
+    if [[ $exitCode == 0 ]]; then return; fi
+    echo "Error: exit code $exitCode from: $task"
+    if [[ $dontExit != 'continue' ]]; then
+        exit $exitCode
+    fi
+}
+
+ensureWeAreUser() {
+    if [[ $(whoami) = 'root' ]]; then
+        echo "Error: must be normal user to run ${0##*/}"
+        exit 2
+    fi
+}
+
+
+function getKeyPair() {
+  #Check for existance of a specific key . If it isn't found good, if it is found exit 2
+  echo "Checking for private keys and concatenated public key using key name: "${HZN_ORG_ID}_${KEY_NAME}
+  if [[ $(/usr/lib/jvm/openjre-11-manual-installation/bin/keytool -list -v -keystore ocs/config/db/v1/creds/owner-keystore.p12 -storepass "${SDO_KEY_PWD}" | grep -E "^Alias name: *${HZN_ORG_ID}_${KEY_NAME}_rsa$") > 0 ]]; then
+    >&2 echo "Key Name Already Used"
+    /usr/lib/jvm/openjre-11-manual-installation/bin/keytool -list -v -keystore ocs/config/db/v1/creds/owner-keystore.p12 -storepass "${SDO_KEY_PWD}" | grep -E "^Alias name: *${HZN_ORG_ID}_${KEY_NAME}_rsa$" >&2
+    #>&2 cat ocs/config/db/v1/creds/publicKeys/${ORG_UNIT}/${ORG_UNIT}_${KEY_NAME}_public-key.pem
+    exit 2
+  else
+    echo "Key Name Not Found"
+  fi
+  }
+
+function allKeys() {
+  for i in "rsa" "ecdsa256" "ecdsa384"; do
+    keyType=$i
+    genKey
+  done
+}
+
+#This function will create a private key that is needed to create a private keystore. Encryption keyType passed will decide which command to run for private key creation
+function genKey() {
+  local privateKey=""${keyType}"private-key.pem"
+  local keyCert=""${keyType}"Cert.crt"
+  #Check if the folder is already created for the keyType (In case of multiple runs)
+  mkdir -p "${keyType}"Key && pushd "${keyType}"Key >/dev/null || return
+  #Generate a private RSA key.
+  if [[ $keyType == "rsa" ]]; then
+    if [ "$(uname)" == "Darwin" ]; then
+      echo "Using macOS, will generate private key and certificate simultaneously."
+      gencmd="openssl req -x509 -nodes -days 3650 -newkey rsa:2048 -keyout "$privateKey" -out "$keyCert""
+    else
+      echo -e "Generating an "${keyType}" private key."
+      openssl genrsa -out "${keyType}"private-key.pem 2048 >/dev/null 2>&1
+      chk $? 'Generating a rsa private key.'
+    fi
+    keyCertGenerator
+  #Generate a private ecdsa (256 or 384) key.
+  elif [[ $keyType == "ecdsa256" ]] || [[ $keyType == "ecdsa384" ]]; then
+    echo -e "Generating an "${keyType}" private key."
+    local var2=$(echo $keyType | cut -f2 -da)
+    if [ "$(uname)" == "Darwin" ]; then
+      echo "Using macOS, will generate private key and certificate simultaneously."
+    else
+      openssl ecparam -genkey -name secp"${var2}"r1 -out "${keyType}"private-key.pem >/dev/null 2>&1
```

Comment:
This comment applies to everywhere you use `>/dev/null 2>&1`: this also throws away any error msg from the cmd, and you'll be left with just the short string `chk` prints. That could make it hard to debug a user problem. Since this script is being run by your golang code (as opposed to directly by the user), you have the luxury of keeping these error msgs, because in your golang code you can preface it with something easier for the user to understand, e.g. "error creating the private key. Details: "

So i suggest you discard stdout and keep stderr using just `>/dev/null`